### ISSUE TITLE
Installation fails

### ISSUE THREAD

**Issue by LeonhardScheck** [pytroll/satpy#11]
Hi,
when I try to install satpy using "pip install satpy" I get the error
"Collecting satpy
  Could not find a version that satisfies the requirement satpy (from versions: )
No matching distribution found for satpy"

It seems that https://pypi.python.org/pypi/satpy/2.0.0a1 contains only meta data, not the actual package...

When I download the package directly from github and do a "python setup.py install" I get
"[...]
Processing dependencies for satpy==2.0.0a1
Searching for pykdtree
Reading https://pypi.python.org/simple/pykdtree/
error: [Errno 2] No such file or directory"

It seems that handling the package dependencies does not work correctly.
So I have installed the following packages with pip: pykdtree trollimage trollsift pyproj

Now I can run "python setup.py install" and successfully "import satpy", but when I try to access SEVIRI data using
global_scene = Scene(platform_name="Meteosat-10", sensor="seviri", start_time=..., base_dir=...)
I get the following error:
ImportError: Could not import reader class 'satpy.readers.mipp_xrit.XritReader' for reader 'mipp_xrit': No module named satin.helper_functions

How can this be fixed?

P.S.: I used the Anaconda Python distribution under SuSE Linux x86_64.

**Comment by davidh-ssec** [comment]
Hi @LeonhardScheck. SatPy is still in early development so we appreciate your patience with getting it installed. The best option for installing right now would be to install from the github repository's `feature-yaml` branch. I'm guessing you installed from `develop` which had some bugs in it that have been fixed in `feature-yaml`. The `feature-yaml` branch is a major rewrite of things we had before and we are almost ready to merge it back to develop.

If you have trouble installing it from that branch let me know. I talked to @mraspaud who is more familiar with the "seviri" reader and he said there may be some configuration files needed that aren't currently included in the package. He'll make those changes in the next day or so and will update this ticket.

**Comment by LeonhardScheck** [comment]
Thanks for the quick reply!

Installing the feature-yaml branch with
`git clone -b feature-yaml https://github.com/pytroll/satpy.git
cd satpy
python setup.py install`
worked without problems...

There are indeed still some configuration files missing (e.g. msg3.cfg), which prevents me from loading data. I will try it again after the next update...

**Comment by mraspaud** [comment]
@LeonhardScheck Hi ! I have just pushed a patch adding the MSG3.cfg file, so that reading MSG3 data should work. You might need the latest mipp (pre-master branch) installed though.

**Comment by LeonhardScheck** [comment]
Hi @mraspaud ! Thanks for the update!
I installed mipp/pre-master, reinstalled satpy/feature-yaml and now I get the following error:

In [4]: global_scene = Scene(platform_name="Meteosat-10", sensor="seviri", start_time=..., base_dir=...)
---------------------------------------------------------------------------
/usr/users/leo.scheck/install/anaconda/lib/python2.7/site-packages/satpy-2.0.0a1-py2.7.egg/satpy/scene.pyc in __init__(self, filenames, ppp_config_dir, reader, base_dir, **info)
    123             reader_instances = finder(reader=reader,
    124                                       sensor=self.info.get("sensor"),
--> 125                                       filenames=filenames)
    126         except ValueError as err:
    127             if filenames is None and base_dir is None:
/usr/users/leo.scheck/install/anaconda/lib/python2.7/site-packages/satpy-2.0.0a1-py2.7.egg/satpy/readers/__init__.pyc in __call__(self, filenames, sensor, reader)
    311                     remaining_filenames)))
    312         if not reader_instances:
--> 313             raise ValueError("No supported files found")
    314         return reader_instances
    315 
ValueError: No supported files found

**Comment by mraspaud** [comment]
@LeonhardScheck It seems to work for me. Could you switch on debuging and paste your log here ?
```from satpy.utils import debug_on
debug_on()```

**Comment by LeonhardScheck** [comment]
Hi @mraspaud ! Finally I had time to look into this again. I started from scratch and got it running, but only after fixing some things:
I had to change "area_defs_to_extent" to "area_def_names_to_extent" and "satpy.satin.helper_functions" to "mpop.satin.helper_functions" in mipp_xrit.py.
And I had to copy MSG3.cfg from the source directory to .../lib/python2.7/site-packages/satpy-2.0.0a1-py2.7.egg/etc/
Thanks for the help!

P.S.: Here is a summary of what I did to install satpy in a fresh virtualenv environment:
pip install numpy pykdtree trollimage trollsift pyproj pyaml h5py netCDF4 pygac h5netcdf
git clone -b pre-master https://github.com/pytroll/mipp.git
cd mipp; python setup.py install; cd ..
git clone -b pre-master https://github.com/pytroll/mpop.git
cd mpop; python setup.py install; cd ..
git clone -b feature-yaml https://github.com/pytroll/satpy.git
cd satpy; python setup.py install; cd ..

**Comment by mraspaud** [comment]
Interesting, thanks for the information. I will try to fix that further so that it works out of the box.

**Comment by davidh-ssec** [comment]
@mraspaud @LeonhardScheck We've made quite a few changes to satpy's develop branch recently. Can this be closed?

**Comment by LeonhardScheck** [comment]
For me the installation works now,  "pip install satpy" was enough. Great!

**Comment by caracicatrz** [comment]
Neither pip install nor the conda forge doesn't work for me. Get the following message

"  Running setup.py clean for numcodecs
Failed to build pykdtree numcodecs
Installing collected packages: pykdtree, numcodecs, zarr, configobj, pyresample, satpy
    Running setup.py install for pykdtree ... error"

Is there a way get over this issue?

**Comment by djhoese** [comment]
@caracicatrz Please create a new issue and include information like your operating system and the version of python you are using. This looks like pykdtree is failing to install which is something the pykdtree developers are working on (CC @mraspaud ). However, installing pykdtree/satpy from conda-forge *should* get around these issues. If you are still having issues though please file a new bug report.
